diff --git a/index.js b/index.js
index c3e5487..1ffb31e 100644
--- a/index.js
+++ b/index.js
@@ -1,6 +1,11 @@
 var Crypto = require('crypto');
 
-var g_PubkeySystem = require('fs').readFileSync(__dirname + '/system.pem');
+var g_PubkeySystem = Buffer.from(`-----BEGIN PUBLIC KEY-----
+MIGdMA0GCSqGSIb3DQEBAQUAA4GLADCBhwKBgQDf7BrWLBBmLBc1OhSwfFkRf53T
+2Ct64+AVzRkeRuh7h3SiGEYxqQMUeYKO6UWiSRKpI2hzic9pobFhRr3Bvr/WARvY
+gdTckPv+T1JzZsuVcNfFjrocejN1oWI0Rrtgt4Bo+hOneoo3S57G9F1fOpn5nsQ6
+6WOiu4gZKODnFMBCiQIBEQ==
+-----END PUBLIC KEY-----`);
 
 /**
  * Verifies a signature using the Steam "System" public key.
@@ -73,12 +78,16 @@ exports.symmetricEncryptWithHmacIv = function(input, key) {
 exports.symmetricDecrypt = function(input, key, checkHmac) {
 	var aesIv = Crypto.createDecipheriv('aes-256-ecb', key, '');
 	aesIv.setAutoPadding(false);
-	aesIv.end(input.slice(0, 16));
-	var iv = aesIv.read();
+	// Synchronously compute the IV
+	var iv = Buffer.concat([
+		aesIv.update(input.subarray(0, 16)), 
+		aesIv.final()
+	]);
 
 	var aesData = Crypto.createDecipheriv('aes-256-cbc', key, iv);
-	aesData.end(input.slice(16));
-	var plaintext = aesData.read();
+	// Synchronously decrypt data
+	var plaintext = aesData.update(input.subarray(16));
+	plaintext = Buffer.concat([plaintext, aesData.final()]);
 
 	if (checkHmac) {
 		// The last 3 bytes of the IV are a random value, and the remainder are a partial HMAC
